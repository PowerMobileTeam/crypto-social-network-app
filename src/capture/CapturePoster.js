import React, { Component } from 'react';
import {
  StyleSheet,
  View,
  ScrollView,
  TextInput,
  Text,
  Alert,
  Button,
  TouchableHighlight,
  TouchableOpacity,
  ActivityIndicator
} from 'react-native';

import { observer, inject } from 'mobx-react/native';
import { Icon } from 'react-native-elements'
import {
  NavigationActions
} from 'react-navigation';

import colors from '../styles/Colors';

import CaptureGallery from './CaptureGallery';
import CapturePreview from './CapturePreview';

import Util from '../common/helpers/util';
import RichEmbedService from '../common/services/rich-embed.service';
import CaptureMetaPreview from './CaptureMetaPreview';
import CapturePostButton from './CapturePostButton';
import { CommonStyle } from '../styles/Common';
import Colors from '../styles/Colors';
import CapturePosterFlags from './CapturePosterFlags';
import UserAutocomplete from '../common/components/UserAutocomplete';

@inject('user', 'capture')
@observer
export default class CapturePoster extends Component {

  /**
   * Disable navigation bar
   */
  static navigationOptions = ({ navigation }) => ({
    headerRight: navigation.state.params && navigation.state.params.headerRight
  });

  state = {
    text: '',
    postImageUri: '',
    hasRichEmbed: false,
    richEmbedUrl: '',
    meta: null,
    mature: false,
    share: {},
    lock: null,
    selection: {
      start:0,
      end: 0
    }
  };


  _RichEmbedFetchTimer;

  /**
   * On component will mount
   */
  componentWillMount() {
    const { setParams } = this.props.navigation;
    setParams({ headerRight: <CapturePostButton onPress={() => this.submit()} /> });
  }

  /**
   * On component did mount
   */
  componentDidMount() {
    const { params } = this.props.navigation.state;
    if (params ) {
      if (params.text) {
        this.setState({text: params.text});
      } else if (params.image) {
        this.onAttachedMedia({
          type: 'image/jpeg',
          uri: params.image
        });
      } else if (params.video) {
        this.onAttachedMedia({
          type: 'video/mp4',
          uri: params.video
        });
      }
    }
  }

  /**
   * On component will unmount
   */
  componentWillUnmount() {
    if (this._RichEmbedFetchTimer) {
      clearTimeout(this._RichEmbedFetchTimer);
    }

    this.deleteAttachment();
  }

  showContext () {
    let group = this.props.navigation.state.params? this.props.navigation.state.params.group : null;
    return group? <Text style={styles.title}> { '( Posting in ' + group.name + ')'} </Text> :null;
  }

  /**
   * Nav to newsfeed
   * @param {object} entity
   */
  navToNewsfeed(entity) {
    const dispatch = NavigationActions.navigate({
      routeName: 'Newsfeed',
      params: {
        prepend: entity,
      },
    });

    this.props.navigation.dispatch(dispatch);
  }

  /**
   * Nav to group
   */
  navToGroup(group, entity) {

    const {state, dispatch, goBack} = this.props.navigation;

    const params = {
      group: group,
      prepend: entity,
    };

    dispatch(NavigationActions.setParams({
      params,
      key: state.params.parentKey, // passed from index
    }));

    goBack(null);

  }

  /**
   * On tag selected in the autocomplete
   */
  onSelectTag = (text) => {
    this.setText(text);
  }

  /**
   * Set the state with cursor position
   */
  onSelectionChanges = (event) => {
    this.setState({selection: event.nativeEvent.selection});
  }

  /**
   * Render
   */
  render() {
    const attachment = this.props.capture.attachment;
    const navigation = this.props.navigation;

    return (
      <View style={{flex:1}}>
        <ScrollView style={styles.posterAndPreviewWrapper}>
          {this.showContext()}
          <View style={styles.posterWrapper} pointerEvents="box-none">
            <TextInput
              style={styles.poster}
              editable={true}
              placeholder='Speak your mind...'
              placeholderTextColor='#ccc'
              underlineColorAndroid='transparent'
              onChangeText={this.setText}
              value={this.state.text}
              multiline={true}
              selectTextOnFocus={false}
              onSelectionChange={this.onSelectionChanges}
            />
          </View>

          {(this.state.meta || this.state.metaInProgress) && <CaptureMetaPreview
            meta={this.state.meta}
            inProgress={this.state.metaInProgress}
            onRemove={this.clearRichEmbedAction}
          />}

          <CapturePosterFlags
            matureValue={this.state.mature}
            shareValue={this.state.share}
            lockValue={this.state.lock}
            onMature={this.onMature}
            onShare={this.onShare}
            onLocking={this.onLocking}
          />

          {attachment.hasAttachment && <View style={styles.preview}>
            <CapturePreview
              uri={attachment.uri}
              type={attachment.type}
            />
            <Icon raised name="md-close" type="ionicon" color='#fff' size={22} containerStyle={styles.deleteAttachment} onPress={() => this.deleteAttachment()}/>
          </View>}

          <CaptureGallery
            onSelected={this.onAttachedMedia}
          />
        </ScrollView>
        <UserAutocomplete
          text={this.state.text}
          selection={this.state.selection}
          onSelect={this.onSelectTag}
        />
      </View>
    );
  }

  /**
   * Attach Media
   */
  onAttachedMedia = async (response) => {
    const attachment = this.props.capture.attachment;

    try {
      const result = await attachment.attachMedia(response);
    } catch(err) {
      console.error(err);
      Alert.alert('caught upload error');
    }
  }

  /**
   * Delete attachment
   */
  async deleteAttachment() {
    const attachment = this.props.capture.attachment;
    // delete only if it has an attachment
    if (attachment.hasAttachment) {
      const result = await attachment.delete();
      if (result === false) Alert.alert('caught error deleting the file');
    }
  }

  /**
   * Submit
   */
  async submit() {
    const attachment = this.props.capture.attachment;

    if (attachment.hasAttachment && attachment.uploading) {
      Alert.alert('Please try again in a moment.');
      return false;
    }

    if (
      !attachment.hasAttachment &&
      !this.state.text &&
      (!this.state.meta || !this.state.meta.perma_url)
    ) {
      Alert.alert('Nothing to post...');
      return false;
    }

    let newPost = {
      message: this.state.text,
      mature: this.state.mature ? 1 : 0,
      wire_threshold: this.state.lock
    };

    if (attachment.guid) {
      newPost.attachment_guid = attachment.guid;
      newPost.attachment_license = attachment.license;
    }

    for (let network in this.state.share) {
      if (this.state.share[network]) {
        newPost[network] = 1;
      }
    }

    if (this.state.meta) {
      newPost = Object.assign(newPost, this.state.meta);
    }

    if (this.props.navigation.state.params && this.props.navigation.state.params.group) {
      newPost.container_guid = this.props.navigation.state.params.group.guid;
    }

    try {
      let response = await this.props.capture.post(newPost);

      if (this.props.reset) {
        this.props.reset();
      }

      // clear attachment data
      attachment.clear();

      this.setState({
        text: '',
        meta: null,
        mature: false,
        share: {},
        lock: null,
      });

      this.props.capture.setPosting(false);

      if (this.props.onComplete) {
        this.props.onComplete(response.entity);
      } else if (this.props.navigation.state.params && this.props.navigation.state.params.group) {
        this.navToGroup(this.props.navigation.state.params.group, response.entity);
      } else {
        this.navToNewsfeed(response.entity);
      }

      return response;
    } catch (e) {
      console.log('error', e);
      Alert.alert('Oooppppss. Looks like there was an error.');
    }
  }

  setText = (text) => {
    this.setState({ text });

    if (this._RichEmbedFetchTimer) {
      clearTimeout(this._RichEmbedFetchTimer);
    }

    setTimeout(this.richEmbedCheck);
  };

  richEmbedCheck = () => {
    const matches = Util.urlReSingle.exec(this.state.text);

    if (matches) {
      const url = (!matches[3] ? 'https://' : '') + matches[0];

      if (
        !this.state.hasRichEmbed ||
        (this.state.hasRichEmbed && url.toLowerCase() !== this.state.richEmbedUrl.toLowerCase())
      ) {
        this._RichEmbedFetchTimer = setTimeout(() => this.setRichEmbed(url), 750);
      }
    }
  };

  clearRichEmbedAction = () => {
    this.clearRichEmbed();

    if (this._RichEmbedFetchTimer) {
      clearTimeout(this._RichEmbedFetchTimer);
    }
  };

  clearRichEmbed() {
    this.setState({
      hasRichEmbed: false,
      richEmbedUrl: '',
      meta: null
    });
  }

  async setRichEmbed(url) {
    this.setState({
      hasRichEmbed: true,
      richEmbedUrl: url,
      meta: null,
      metaInProgress: true
    });

    try {
      const meta = await RichEmbedService.getMeta(url);

      this.setState({ meta, metaInProgress: false })
    } catch (e) {
      this.setState({ metaInProgress: false });
      console.error(e);
    }
  }


  onMature = () => {
    const mature = !this.state.mature;
    this.setState({ mature });
  }

  onShare = network => {
    const share = Object.assign({}, this.state.share);

    if (share[network]) {
      delete share[network];
    } else {
      share[network] = true;
    }

    this.setState({ share });
  }

  onLocking = lock => {
    this.setState({ lock });
  }
}

const styles = StyleSheet.create({
  posterAndPreviewWrapper: {
    backgroundColor: 'white',
    flex:1
  },
  posterWrapper: {
    minHeight: 100,
    flexDirection: 'row',
    backgroundColor: '#FFF',
  },
  title: {
    margin:2,
    padding:4,
    color: '#4b4b4b'
  },
  poster: {
    padding: 12,
    paddingTop: 24,
    flex: 1,
  },
  posterButton: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
    alignContent: 'center',
  },
  preview: {
    flex: 1,
    minHeight: 200,
    flexDirection: 'row',
    alignItems: 'stretch',
    position: 'relative',
  },
  avatar: {
    flex: 1,
  },
  gallery: {
    flex: 1,
    flexDirection: 'column',
    alignItems: 'stretch',
  },
  deleteAttachment: {
    position: 'absolute',
    backgroundColor:'#4690DF',
    width:28,
    height:28,
    right: 8,
    top: 0,
  }
});
