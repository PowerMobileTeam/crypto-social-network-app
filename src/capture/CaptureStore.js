import {
  observable,
  action
} from 'mobx'
import AttachmentStore from '../common/stores/AttachmentStore';
import apiService from '../common/services/api.service';
import { post } from './CaptureService';

/**
 * Capture store
 */
class CaptureStore {
  attachment = new AttachmentStore();

  @observable isPosting = false;

  @observable socialNetworks = {
    facebook: false,
    twitter: false,
  };

  /**
   * Set posting
   * @param {boolean} value
   */
  @action
  setPosting(value) {
    this.isPosting = value;
  }

  /**
   * Reset store's status
   */
  reset() {
    this.isPosting = false;
    this.attachment.clear();
    this.socialNetworks = {
      facebook: false,
      twitter: false,
    };
  }

  /**
   * Post
   * @param {object} newPost
   */
  async post(newPost) {
    this.setPosting(true);
    const result = await post(newPost);
    this.setPosting(false);
    return result;
  }

  /**
   * Load third party social network status
   */
  @action async loadThirdPartySocialNetworkStatus() {
    const response = await apiService.get(`api/v1/thirdpartynetworks/status`);

    if (response && response.thirdpartynetworks) {
      const networks = response.thirdpartynetworks;

      this.socialNetworks.twitter = networks.twitter && networks.twitter.connected;
      this.socialNetworks.facebook = networks.facebook && networks.facebook.connected;
    }
    return response;
  }
}

export default CaptureStore;
