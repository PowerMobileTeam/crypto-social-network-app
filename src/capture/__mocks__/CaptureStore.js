import AttachmentStore from '../../common/stores/__mocks__/AttachmentStore';

const mock = jest.fn().mockImplementation(() => {

  return {
    setPosting: jest.fn(),
    reset: jest.fn(),
    post: jest.fn(),
    loadThirdPartySocialNetworkStatus: jest.fn(),
    attachment: new AttachmentStore(),
    isPosting: false,
    socialNetworks: {
      facebook: false,
      twitter: false,
    }
  };
});

export default mock;
