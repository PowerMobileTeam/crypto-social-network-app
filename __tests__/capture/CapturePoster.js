import 'react-native';
import React from 'react';
import { Alert } from 'react-native';
import { shallow } from 'enzyme';
import { Icon } from 'react-native-elements'

import CapturePoster from '../../src/capture/CapturePoster';
import CapturePreview from '../../src/capture/CapturePreview';
import UserStore from '../../src/auth/UserStore';
import CaptureStore from '../../src/capture/CaptureStore';
import appNavigation from '../../AppNavigation';
// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

jest.mock('../../src/auth/UserStore');
jest.mock('../../src/capture/CaptureStore');
jest.mock('../../src/common/stores/NavigationStore');
jest.mock('../../src/capture/CapturePostButton', () => 'CapturePostButton');
jest.mock('../../src/capture/CapturePosterFlags', () => 'CapturePosterFlags');
jest.mock('../../src/capture/CapturePreview', () => 'CapturePreview');

Alert.alert = jest.fn();

/**
 * Tests
 */
describe('cature poster component', () => {

  let userStore, capture, navigation;

  const paramsVideo = {uri: 'file://video.mp4', type: 'video/mp4'};

  const paramsImage = {uri: 'file://image.jpg', type: 'image/jpeg'};

  beforeEach(() => {
    userStore = new UserStore();
    capture = new CaptureStore();
    navigation = appNavigation.buildNavigator();
    navigation.setParams = jest.fn();
    navigation.state.params = {};
    capture.attachment.attachMedia.mockClear();
  });

  it('should renders correctly', () => {
    const galley = renderer.create(
      <CapturePoster.wrappedComponent
        user={userStore}
        capture={capture}
        navigation={navigation}
      />
    ).toJSON();
    expect(galley).toMatchSnapshot();
  });

  it('should receive text parameters on did mount', () => {
    const spySetState = jest.spyOn(CapturePoster.wrappedComponent.prototype, 'setState');

    navigation.state.params = {text: 'hello'};

    const wrapper = shallow(
      <CapturePoster.wrappedComponent
        user={userStore}
        capture={capture}
        navigation={navigation}
      />
    );

    // should be called
    expect(spySetState).toHaveBeenCalled();

    // should be called only once
    expect(spySetState.mock.calls.length).toBe(1);

    // should be called with text hello
    expect(spySetState.mock.calls[0][0].text).toBe('hello');
  });

  it('should receive image parameters on did mount and attach it', () => {

    navigation.state.params = {image: 'file://image.jpg'};

    const wrapper = shallow(
      <CapturePoster.wrappedComponent
        user={userStore}
        capture={capture}
        navigation={navigation}
      />
    );

    // should be called
    expect(capture.attachment.attachMedia).toHaveBeenCalled();

    // should be called only once
    expect(capture.attachment.attachMedia.mock.calls.length).toBe(1);

    // should be called with the image
    expect(capture.attachment.attachMedia.mock.calls[0][0]).toEqual({type: 'image/jpeg', uri:'file://image.jpg'} );
  });

  it('should receive video parameters on did mount and attach it', () => {

    navigation.state.params = {video: 'file://video.mp4'};

    const wrapper = shallow(
      <CapturePoster.wrappedComponent
        user={userStore}
        capture={capture}
        navigation={navigation}
      />
    );

    // should be called
    expect(capture.attachment.attachMedia).toHaveBeenCalled();

    // should be called only once
    expect(capture.attachment.attachMedia.mock.calls.length).toBe(1);

    // should be called with the video
    expect(capture.attachment.attachMedia.mock.calls[0][0]).toEqual({type: 'video/mp4', uri:'file://video.mp4'} );
  });

  it('should show the preview when an image is attached', async (done) => {
    try {


      const wrapper = shallow(
        <CapturePoster.wrappedComponent
          user={userStore}
          capture={capture}
          navigation={navigation}
        />
      );

      // emulate image attachment
      capture.attachment.hasAttachment = true;
      capture.attachment.uri = paramsImage.uri;
      capture.attachment.type = paramsImage.type;

      // update component
      wrapper.update();

      // find Capture Preview
      const preview = wrapper.dive().find(CapturePreview);

      expect(preview.length).toBe(1);

      done();
    } catch (e) {
      done.fail(e);
    }
  });

  it('should show the preview when a video is attached', async (done) => {
    try {

      const wrapper = shallow(
        <CapturePoster.wrappedComponent
          user={userStore}
          capture={capture}
          navigation={navigation}
        />
      );

      // emulate video attachment
      capture.attachment.hasAttachment = true;
      capture.attachment.uri = paramsVideo.uri;
      capture.attachment.type = paramsVideo.type;

      // update component
      wrapper.update();

      // find Capture Preview
      const preview = wrapper.dive().find(CapturePreview);

      expect(preview.length).toBe(1);

      done();
    } catch (e) {
      done.fail(e);
    }
  });

  it('should call attachment store delete when user tap delete', async (done) => {
    try {

      // emulate video attachment
      capture.attachment.hasAttachment = true;
      capture.attachment.uri = paramsVideo.uri;
      capture.attachment.type = paramsVideo.type;

      const wrapper = shallow(
        <CapturePoster.wrappedComponent
          user={userStore}
          capture={capture}
          navigation={navigation}
        />
      );

      // find delete icon
      const icon = wrapper.dive().find(Icon);

      expect(icon.length).toBe(1);

      // simulate press on image
      icon.at(0).simulate('press');

      // should be called
      expect(capture.attachment.delete).toHaveBeenCalled();

      // should be called only once
      expect(capture.attachment.delete.mock.calls.length).toBe(1);

      done();
    } catch (e) {
      done.fail(e);
    }
  });

  it('should call attachment store delete when user leave', async (done) => {
    try {

      // emulate video attachment
      capture.attachment.hasAttachment = true;
      capture.attachment.uri = paramsVideo.uri;
      capture.attachment.type = paramsVideo.type;

      const wrapper = shallow(
        <CapturePoster.wrappedComponent
          user={userStore}
          capture={capture}
          navigation={navigation}
        />
      );

      // unmount
      await wrapper.instance().componentWillUnmount();

      // should be called
      expect(capture.attachment.delete).toHaveBeenCalled();

      // should be called only once
      expect(capture.attachment.delete.mock.calls.length).toBe(1);

      done();
    } catch (e) {
      done.fail(e);
    }
  });

  it('should return false if the user post before finish an upload', async (done) => {
    try {

      // emulate video attachment
      capture.attachment.hasAttachment = true;
      capture.attachment.uri = paramsVideo.uri;
      capture.attachment.type = paramsVideo.type;
      capture.attachment.uploading = true;

      const wrapper = shallow(
        <CapturePoster.wrappedComponent
          user={userStore}
          capture={capture}
          navigation={navigation}
        />
      );

      // submit
      const result = await wrapper.instance().submit();

      // should be false
      expect(result).toBe(false);

      done();
    } catch (e) {
      done.fail(e);
    }
  });

  it('should return false if there is nothing to post', async (done) => {
    try {
      const wrapper = shallow(
        <CapturePoster.wrappedComponent
          user={userStore}
          capture={capture}
          navigation={navigation}
        />
      );

      // submit
      const result = await wrapper.instance().submit();

      // should be false
      expect(result).toBe(false);

      done();
    } catch (e) {
      done.fail(e);
    }
  });

  it('should call store\'s post on submit and return the response', async (done) => {
    try {
      const wrapper = shallow(
        <CapturePoster.wrappedComponent
          user={userStore}
          capture={capture}
          navigation={navigation}
        />
      );

      wrapper.instance().setText('some awesome post');

      const response = {entity:{guid:1, title: 'some awesome post'}};

      capture.post.mockResolvedValue(response);

      // submit
      const result = await wrapper.instance().submit();

      // should be called
      expect(capture.post).toHaveBeenCalled();

      // should be called only once
      expect(capture.post.mock.calls.length).toBe(1);

      expect(capture.post.mock.calls[0][0]).toEqual({mature: 0, message: "some awesome post", wire_threshold: null});

      expect(result).toEqual(response)

      done();
    } catch (e) {
      done.fail(e);
    }
  });

  it('should call onComplete after submit', async (done) => {
    try {
      const onCompleteMock = jest.fn();

      const wrapper = shallow(
        <CapturePoster.wrappedComponent
          user={userStore}
          capture={capture}
          navigation={navigation}
          onComplete={onCompleteMock}
        />
      );

      wrapper.instance().setText('some awesome post');

      const response = {entity:{guid:1, title: 'some awesome post'}};

      capture.post.mockResolvedValue(response);

      // submit
      const result = await wrapper.instance().submit();

      // should be called
      expect(onCompleteMock).toHaveBeenCalled();

      done();
    } catch (e) {
      done.fail(e);
    }
  });

  it('should add third party networks share to post', async (done) => {
    try {

      const wrapper = shallow(
        <CapturePoster.wrappedComponent
          user={userStore}
          capture={capture}
          navigation={navigation}
        />
      );

      wrapper.instance().setText('some awesome post');
      wrapper.instance().onShare('facebook');
      wrapper.instance().onShare('twitter');

      const response = {entity:{guid:1, title: 'some awesome post'}};

      capture.post.mockResolvedValue(response);

      // submit
      const result = await wrapper.instance().submit();

      // should be called
      expect(capture.post).toHaveBeenCalled();

      // should be called only once
      expect(capture.post.mock.calls.length).toBe(1);

      expect(capture.post.mock.calls[0][0]).toEqual({
        mature: 0,
        message: "some awesome post",
        wire_threshold: null,
        facebook: 1,
        twitter: 1
      });

      done();
    } catch (e) {
      done.fail(e);
    }
  });

  it('should call store\'s post with attachment guid', async (done) => {
    try {
      const wrapper = shallow(
        <CapturePoster.wrappedComponent
          user={userStore}
          capture={capture}
          navigation={navigation}
        />
      );

        // emulate image attachment
      capture.attachment.hasAttachment = true;
      capture.attachment.guid = 1000;
      capture.attachment.uri = paramsImage.uri;
      capture.attachment.type = paramsImage.type;
      capture.attachment.uploading = false;

      wrapper.instance().setText('some awesome post');

      const response = {entity:{guid:1, title: 'some awesome post'}};

      capture.post.mockResolvedValue(response);

      // submit
      const result = await wrapper.instance().submit();

      // should be called
      expect(capture.post).toHaveBeenCalled();

      // should be called only once
      expect(capture.post.mock.calls.length).toBe(1);

      // should send the attachment data
      expect(capture.post.mock.calls[0][0]).toEqual({
        mature: 0,
        message: "some awesome post",
        wire_threshold: null,
        attachment_guid: 1000,
        attachment_license: ''}
      );
      // should return server response
      expect(result).toEqual(response)

      // should clear the attachment
      expect(capture.attachment.clear).toHaveBeenCalled();

      done();
    } catch (e) {
      done.fail(e);
    }
  });

  it('should alert the user on submit error', async (done) => {
    try {
      const wrapper = shallow(
        <CapturePoster.wrappedComponent
          user={userStore}
          capture={capture}
          navigation={navigation}
        />
      );

      wrapper.instance().setText('some awesome post');

      capture.post.mockResolvedValue(new Error('some error'));

      // submit
      const result = await wrapper.instance().submit();

      // should be called
      expect(Alert.alert).toHaveBeenCalled();

      done();
    } catch (e) {
      done.fail(e);
    }
  });
});