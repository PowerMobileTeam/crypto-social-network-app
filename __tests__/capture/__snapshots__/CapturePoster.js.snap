// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cature poster component should renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RCTScrollView
    style={
      Object {
        "backgroundColor": "white",
        "flex": 1,
      }
    }
  >
    <View>
      <View
        pointerEvents="box-none"
        style={
          Object {
            "backgroundColor": "#FFF",
            "flexDirection": "row",
            "minHeight": 100,
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          editable={true}
          multiline={true}
          onChangeText={[Function]}
          onSelectionChange={[Function]}
          placeholder="Speak your mind..."
          placeholderTextColor="#ccc"
          selectTextOnFocus={false}
          style={
            Object {
              "flex": 1,
              "padding": 12,
              "paddingTop": 24,
            }
          }
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
      <CapturePosterFlags
        lockValue={null}
        matureValue={false}
        onLocking={[Function]}
        onMature={[Function]}
        onShare={[Function]}
        shareValue={Object {}}
      />
      <View>
        <View
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <View
            style={
              Object {
                "height": 125,
              }
            }
          >
            <View
              style={
                Object {
                  "flex": 1,
                  "flexDirection": "row",
                  "paddingLeft": 1,
                  "paddingRight": 1,
                }
              }
            >
              <View
                accessibilityComponentType={undefined}
                accessibilityLabel={undefined}
                accessibilityTraits={undefined}
                accessible={true}
                hasTVPreferredFocus={undefined}
                hitSlop={undefined}
                isTVSelectable={true}
                nativeID={undefined}
                onLayout={undefined}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "alignContent": "center",
                    "alignItems": "center",
                    "backgroundColor": "#FFF",
                    "borderColor": "#ECECEC",
                    "borderWidth": 1,
                    "flex": 1,
                    "flexDirection": "column",
                    "justifyContent": "center",
                    "margin": 1,
                  }
                }
                testID={undefined}
                tvParallaxProperties={undefined}
              >
                <View
                  style={
                    Object {
                      "alignContent": "center",
                      "alignItems": "center",
                      "flexDirection": "column",
                      "justifyContent": "center",
                    }
                  }
                >
                  <Text
                    accessible={true}
                    allowFontScaling={false}
                    ellipsizeMode="tail"
                    style={
                      Array [
                        Object {
                          "color": undefined,
                          "fontSize": 36,
                        },
                        Object {
                          "color": "#444",
                        },
                        Object {
                          "fontFamily": "Ionicons",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                      ]
                    }
                  >
                    
                  </Text>
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Object {
                        "color": "#444",
                        "letterSpacing": 1.25,
                      }
                    }
                  >
                    Gallery
                  </Text>
                </View>
              </View>
              <View
                accessibilityComponentType={undefined}
                accessibilityLabel={undefined}
                accessibilityTraits={undefined}
                accessible={true}
                hasTVPreferredFocus={undefined}
                hitSlop={undefined}
                isTVSelectable={true}
                nativeID={undefined}
                onLayout={undefined}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "alignContent": "center",
                    "alignItems": "center",
                    "backgroundColor": "#FFF",
                    "borderColor": "#ECECEC",
                    "borderWidth": 1,
                    "flex": 1,
                    "flexDirection": "column",
                    "justifyContent": "center",
                    "margin": 1,
                  }
                }
                testID={undefined}
                tvParallaxProperties={undefined}
              >
                <View
                  style={
                    Object {
                      "alignContent": "center",
                      "alignItems": "center",
                      "flexDirection": "column",
                      "justifyContent": "center",
                    }
                  }
                >
                  <Text
                    accessible={true}
                    allowFontScaling={false}
                    ellipsizeMode="tail"
                    style={
                      Array [
                        Object {
                          "color": undefined,
                          "fontSize": 36,
                        },
                        Object {
                          "color": "#444",
                        },
                        Object {
                          "fontFamily": "Ionicons",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                      ]
                    }
                  >
                    
                  </Text>
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Object {
                        "color": "#444",
                        "letterSpacing": 1.25,
                      }
                    }
                  >
                    Photo
                  </Text>
                </View>
              </View>
              <View
                accessibilityComponentType={undefined}
                accessibilityLabel={undefined}
                accessibilityTraits={undefined}
                accessible={true}
                hasTVPreferredFocus={undefined}
                hitSlop={undefined}
                isTVSelectable={true}
                nativeID={undefined}
                onLayout={undefined}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "alignContent": "center",
                    "alignItems": "center",
                    "backgroundColor": "#FFF",
                    "borderColor": "#ECECEC",
                    "borderWidth": 1,
                    "flex": 1,
                    "flexDirection": "column",
                    "justifyContent": "center",
                    "margin": 1,
                  }
                }
                testID={undefined}
                tvParallaxProperties={undefined}
              >
                <View
                  style={
                    Object {
                      "alignContent": "center",
                      "alignItems": "center",
                      "flexDirection": "column",
                      "justifyContent": "center",
                    }
                  }
                >
                  <Text
                    accessible={true}
                    allowFontScaling={false}
                    ellipsizeMode="tail"
                    style={
                      Array [
                        Object {
                          "color": undefined,
                          "fontSize": 36,
                        },
                        Object {
                          "color": "#444",
                        },
                        Object {
                          "fontFamily": "Ionicons",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                      ]
                    }
                  >
                    
                  </Text>
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Object {
                        "color": "#444",
                        "letterSpacing": 1.25,
                      }
                    }
                  >
                    Video
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
        <View
          onLayout={undefined}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "white",
              "flex": 1,
              "justifyContent": "center",
            }
          }
        >
          <ActivityIndicator
            animating={true}
            color="#999999"
            hidesWhenStopped={true}
            size="large"
          />
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
